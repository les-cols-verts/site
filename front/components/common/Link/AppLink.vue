<template>
  <a
    :style="{ backgroundColor: backgroundColor, color: color }"
    class="link"
    :href="href"
  >
    {{ text }}
  </a>
</template>
<script>
export default {
  name: 'AppLink',
  props: {
    color: { type: String, default: () => '' },
    backgroundColor: { type: String, default: () => '' },
    text: { type: String, default: () => '' },
    href: { type: String, default: () => '/' },
  },
};
</script>
<style scoped>
.link {
  background-color: #0d8544;
  color: #fff;
  margin: 1em;
  padding: 1em 2em;
  font-weight: bold;
  text-decoration: none;
  border-radius: 2em;
}

.link:hover,
.link:focus {
  background-color: #ffbb02;
}
</style>
