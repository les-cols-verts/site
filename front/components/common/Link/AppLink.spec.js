import AppLink from './AppLink.vue';

describe('AppLink', () => {
  describe('template', () => {
    it('should match snapshot', () => {
      // When
      const wrapper = shallowMount(AppLink);

      // Then
      expect(wrapper).toMatchSnapshot();
    });

    it('should have disabled css attributes when button is disabled', () => {
      // Given
      const propsData = {
        isDisabled: true,
        color: 'blue',
        backgroundColor: 'red',
        text: 'text',
        title: 'title',
      };

      // When
      const wrapper = shallowMount(AppLink, { propsData });

      // Then
      expect(wrapper).toMatchSnapshot();
    });
  });
});
