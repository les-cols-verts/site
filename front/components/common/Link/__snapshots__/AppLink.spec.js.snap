// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLink template should have disabled css attributes when button is disabled 1`] = `
<a href="/" class="link" style="background-color: red; color: blue;" isdisabled="true" title="title">
  text
</a>
`;

exports[`AppLink template should match snapshot 1`] = `
<a href="/" class="link">

</a>
`;
